client/app/pages/alert/components/NotificationTemplate.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { head } from 'lodash';
+import Mustache from 'mustache';
+
+import { HelpTrigger } from '@/components/HelpTrigger';
+import { Alert as AlertType, Query as QueryType } from '@/components/proptypes';
+
+import Input from 'antd/lib/input';
+import Select from 'antd/lib/select';
+import Modal from 'antd/lib/modal';
+import Switch from 'antd/lib/switch';
+
+import './NotificationTemplate.less';
+
+
+function normalizeCustomTemplateData(alert, query, columnNames, resultValues) {
+  const topValue = resultValues && head(resultValues)[alert.options.column];
+
+  return {
+    ALERT_STATUS: 'TRIGGERED',
+    ALERT_EVALUATION_TIME: new Date(),
+    ALERT_CONDITION: alert.options.op,
+    ALERT_THRESHOLD: alert.options.value,
+    ALERT_NAME: alert.name,
+    ALERT_URL: `${window.location.origin}/alerts/${alert.id}`,
+    QUERY_NAME: query.name,
+    QUERY_URL: `${window.location.origin}/queries/${query.id}`,
+    QUERY_RESULT_VALUE: topValue,
+    QUERY_RESULT_ROWS: resultValues,
+    QUERY_RESULT_COLS: columnNames,
+  };
+}
```

Comment:
@arikfr this is a duplicate of the server side implementation and that's not good practice.
Wdyt of making an api call that sends a template string and returns the rendered string.

Reply:
I noticed this too, but decided not to burden you with more work :) 

Ironically, in the original implementation there was an API call, but then we switched to Mustache so we can render it in the client... But I agree that we should have an endpoint for it.

Reply:
I'll leave as is and open an issue once this is merged.